kernel/src/process/program_loader/elf/elf_file.rs

```diff
@@ -9,136 +10,174 @@ use xmas_elf::{
 use crate::{
     fs::utils::{Inode, PATH_MAX},
     prelude::*,
+    vm::{perms::VmPerms, vmar::VMAR_CAP_ADDR},
 };
+
+/// A wrapper for the [`xmas_elf`] ELF parser.
 pub struct ElfHeaders {
-    pub elf_header: ElfHeader,
-    pub program_headers: Vec<ProgramHeader64>,
+    elf_header: ElfHeader,
+    load_phdrs: Vec<ElfLoadPhdr>,
```

Comment:
Would it be better to rename this field to `loadable_phdrs` and the type to `ElfLoadablePhdr`? If `ElfLoadablePhdr` feels too verbose, we could simply drop the `Elf` prefix.